src/screens/MainApp/DetailedDiagnose/Carousel/index.js

```diff
@@ -0,0 +1,77 @@
+import React, { useRef, useState, useEffect } from 'react'
+import { Image, View, ActivityIndicator } from 'react-native'
+import Carousel, { Pagination } from 'react-native-snap-carousel'
+import { getURL } from '~/mixins/storage'
+import DrCannabisLogo from '~/assets/images/DrCannabis.png'
+import styles, { CAROUSEL_SLIDER_WIDTH, ITEM_WIDTH, ITEM_HEIGHT } from './styles'
+
+const CarouselImage = ({ uri }) => {
+  return (
+    <View style={styles.carouselImageContainer}>
+      <Image
+        resizeMethod='scale'
+        style={uri ? styles.carouselImage : styles.placeholder}
+        source={uri ? { uri } : DrCannabisLogo}
+      />
+    </View>
+  )
+}
+
+const MyCarousel = ({ images }) => {
+  const [activeIndex, setActiveIndex] = useState(0)
+  const [fetchingURLs, setFetchingURLs] = useState(true)
+  const [imageExternalURLs, setImageExternalURLs] = useState([])
+  const carouselRef = useRef()
+
+  useEffect(() => {
+    const fetchImagesURLs = async () => {
+      const urls = []
+      setFetchingURLs(true)
+
+      await Promise.all(
+        images.map(async (imageUID, index) => {
+          const url = await getURL(imageUID)
+          urls.push(url)
+        })
+      )
+
+      setImageExternalURLs(urls)
+      setFetchingURLs(false)
```

Comment:
If something fails, this statement will not execute and the state won't change to `fetchingURLs=false`. Consider wrapping the `await Promise.all` in a `try/catch/finally` block, and put this inside `finally`.

Reply:
`getURL` returns the real URL, but if it fails, returns null (the carousel then shows a placeholder image).

I do not think there is something else that could fail but I'l will add the `try/catch` for safety